include/picongpu/plugins/openPMD/openPMDWriter.hpp

```diff
@@ -972,6 +973,79 @@ make sure that environment variable OPENPMD_BP_BACKEND is not set to ADIOS1.
                 dumpData(currentStep);
             }
 
+            /** Checks if an loaded openPMD series is supported for restarting.
+             *
+             * The openPMD series attribute 'picongpuIOVersionMajor', 'picongpuIOVersionMinor', and the current
+             * PIConGPU IO version is compared to find incompatible versions. In case the loaded series is incompatible
+             * an runtime error is thrown.
+             *
+             * @attention Loading a series with older IO major file version as the current PIConGPU IO version is only
+             * possible if it is explicit allowed by this function e.g. in cases the IO plugin is guaranteeing the
+             * compatibility. Loading a file with a newer IO major version is not possible.
+             *
+             * @param series OpenPMD series which is used to load restart data.
+             */
+            void checkIOFileVersionRestartCompatibility(std::unique_ptr<::openPMD::Series>& series) const
+            {
+                /* Major version 0 and not having the attribute picongpuIOVersionMajor is handled equally later on.
+                 * Version 0 will not get any minor version change in the future. The first possible version with
+                 * minor not zero will be file IO version 1.1.
+                 */
+                int ioVersionUsedFileMajor = 0;
+                int ioVersionUsedFileMinor = 0;
+
+                try
+                {
+                    ioVersionUsedFileMajor = series->getAttribute("picongpuIOVersionMajor").get<int>();
+                }
+                catch(std::runtime_error const&)
```

Comment:
Why don't we catch everything? The handler seems not specific to a `std::` exception type.

Same in the `catch` below.

Reply:
I would like to catch `openPMD::no_such_attribute_error` but somehow this is not visible within PIConGPU. I would like to catch only one specific error in cases where the backend throws other error types too. Catching everything will maybe hide different errors from the backend.

Reply:
Oh, so here you are using exceptions just to check if the attribute exists? I think that is confusing and perhaps not the best mechanism for this task. But if there is no other way, fine.

Reply:
I missed that there is a `openPMD::containsAttribute()` available. I will use this function for the check.

Reply:
I updated the code and check now for the attribute before loading.